### PR TITLE
Add custom css modified from @types/jss/css.d.ts to allow jss-expand …

### PR DESCRIPTION
**Pull request by amwill04**
`jss-expand` plugin was not being correctly typed. 

```
{
    padding: [10, 10] // failed
    margin: {
        top: 10 // failed
    }
}
```

Used `typescript 2.9` `Exclude` to omit `jss-expand` types and re-write them. Have used the types from `@types/jss` but probable the source in `jss` should be updated independently.

- [x ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ x] Test the change in your own code. (Compile and run.)
- [x ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [ x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
~- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>~
~- [ ] Increase the version number in the header if appropriate.~
~- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.~

### CODE REVIEW

**Comment by amcasey** [comment]
@eps1lon are you interested in reviewing this change?

**Review by amcasey — CHANGES_REQUESTED** [review]

File: types/react-jss/package.json [hunk 1]
```diff
@@ -1,6 +1,8 @@
 {
   "private": true,
   "dependencies": {
-    "csstype": "^2.0.0"
+    "csstype": "^2.0.0",
+    "indefinite-observable": "^1.0.1",
+    "typescript": "^3.0.3"
```
[the remark this answers is not in the document]
Comment (amwill04) [reply types/react-jss/package.json]: Sorry, dont have typescript installed globally so added it to run `tsc` from the package directory.

Will remove the dependency - is there a best practice? All the documentation said was `Otherwise, run tsc in the package directory`

[the remark this answers is not in the document]
Comment (amcasey) [reply types/react-jss/package.json]: There may be a trick for having a local installation without a package.json entry but I, unfortunately, am not enough of an npm expert to know it.

**Comment by amcasey** [comment]
Looks like you left in the comment before the dependency you deleted.

**Review by eps1lon — CHANGES_REQUESTED** [review]
It's really hard to review if you mix in formatting changes.

I think this patch should be applied to `jss`. `react-jss` is just a binding.

**Comment by amwill04** [comment]
@eps1lon 

apologies on the formatting.

If i update the underlying `css` type in `jss` then are you happy for it to be updated to read in from the `Style` type as currently `react-jss` would still fail on `jss-expand` plugin witch is available by default in `react-jss`

**Comment by eps1lon** [comment]
TL;DR: `import { JssExpand } from 'jss'`;

I ran into the same issue a few days ago and practically gave up. A runtime plugin system is a real nightmare for definitions with implicit state.

In a perfect world you would construct the accepted `Style` object from the options. I thinks it's ok for now to accept `Style` based on the default options.

This is the approach I would prefer:
- create definitions for `jss-expand` which only include the shape of the accepted `Style` object (only the additional properties)
- think of a way how to compose the basic `Style` shape with the shape from a plugin
- use that enhanced shape in `react-jss`

It would be really nice if you could make this work. I'm only asking this because I'm only using `jss` and not `react-jss` and I would like to use that feature too.

**Comment by orta** [comment]
This PR hasn't been touched in over a year, it's quite likely that these changes could have been upstreamed into react-jss now given the merge conflicts. If it hasn't I'd recommend rebasing with a new PR.
